File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 1, i = 1;
        while(i < nums.length) {
            if(nums[i] != nums[i - 1]) {
                nums[k++] = nums[i];
            }
            i++;
        }
        return k;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-2.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n = len(nums) - 1

        bits = 1
        for _ in range(n):
            bits = bits << 1
            bits = bits | 1
        
        for num in nums:
            bit = 1 << (num - 1)
            nbit = bits & bit
            if not nbit:
                return num
            bits = bits ^ bit

        return 0